File: 027.LeetCode@167_Two_Sum_II_Input_Array_Is_Sorted/027.LeetCode@167_Two_Sum_II_Input_Array_Is_Sorted.py
class Solution:
    def twoSum(self, numbers, target):
        startidx=0
        endidx=len(numbers)-1
        while startidx<endidx:
            if numbers[startidx]+numbers[endidx]==target:
                return [startidx+1,endidx+1]
            elif numbers[startidx]+numbers[endidx]>target:
                endidx -= 1
            else:
                startidx += 1

File: 027.LeetCode@167_Two_Sum_II_Input_Array_Is_Sorted/027.LeetCode@167_Two_Sum_II_Input_Array_Is_Sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int startidx=0, endidx=numbers.length-1;
        while(startidx<endidx){
            int sum=numbers[startidx]+numbers[endidx];
            if(sum==target){
                return new int[]{startidx+1, endidx+1};
            }
            else if(sum>target){
                endidx--;
            }
            else{
                startidx++;
            }
        }
        return new int[]{-1,-1};
    }
}